Mathlib/AlgebraicTopology/Homotopy.lean

```diff
@@ -0,0 +1,263 @@
+/-
+Copyright (c) 2023 Joseph Tooby-Smith. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Tooby-Smith
+-/
+
+import Mathlib.AlgebraicTopology.SimplicialObject
+import Mathlib.AlgebraicTopology.Quasicategory
+import Mathlib.CategoryTheory.Limits.Presheaf
+import Mathlib.CategoryTheory.Limits.Shapes.Types
+import Mathlib.CategoryTheory.Yoneda
+import Mathlib.Data.Fin.VecNotation
+import Mathlib.Tactic.FinCases
+
+/-! # Homotopies in quasi-categories
+
+A homotopy in a quasi-category is a 2-simplex such that the 0-edge is the identity.
+
+We define homotopies and show that they form an equivelence relation.
+
+## TODO
+* define the homotopy category
+
+-/
+
+namespace Quasicategory
+
+open CategoryTheory Simplicial
+open SimplexCategory Finset Opposite
+open SSet
+/-- A homotopy is a 2-simplex τ satisfying certain conditions on its faces.  -/
+class homotopy  {C : SSet} [Quasicategory C] (f g : C _[1]) (τ  : C _[2]) : Prop where
+    prop_δ₂ : (C.map (δ 2).op) τ =f
+    prop_δ₁ : (C.map (δ 1).op) τ =g
+    prop_δ₀ : (C.map (δ 0).op) τ =(C.map (σ 0).op) ((C.map (δ 0).op) f)
+
+/--Two 1-simplices are `homotopic` iff there exists a homotopy between them.-/
+def homotopic {C : SSet} [Quasicategory C] (f g : C _[1]) : Prop := ∃ (σ : C _[2]), homotopy f g σ
```

Comment:
There are ways to take more advantage of the Lean type theory in order to get a better design of homotopies (and this would be consistent with the design of homotopies between morphisms of homological complexes), something like:
```lean
universe u

variable {C : SSet.{u}} [Quasicategory C]

structure Homotopy (f g : C _[1]) where
  τ : C _[2]
  δ₂ : C.δ 2 τ = f
  δ₁ : C.δ 1 τ = g
  δ₀ : C.δ 0 τ = C.σ 0 (C.δ 0 f)

def Homotopic (f g : C _[1]) : Prop := Nonempty (Homotopy f g)
```
Then, you may introduce a definition `Homotopy.refl`, and lemmas `Homotopic.refl/symm/trans`.